test_rclcpp/CMakeLists.txt

```diff
@@ -66,6 +66,16 @@ if(BUILD_TESTING)
       "rclcpp")
   endfunction()
 
+  function(custom_gtest_executable target)
+    ament_add_gtest_executable(${target} ${ARGN})
+    target_compile_definitions(${target}
+      PUBLIC "RMW_IMPLEMENTATION=${rmw_implementation}")
```

Comment:
`RMW_IMPLEMENTATION` as a compile definition works? I thought it had to be an environment variable. Test communication uses an environment variable

https://github.com/ros2/system_tests/blob/394f326f2b3da52d802dad1090cddd8ef6fba5b7/test_communication/CMakeLists.txt#L94-L102

Reply:
This is a direct copy of `custom_executable` except that it uses `ament_add_gtest_executable` instead of `add_executable`, so I don't know but that seems to be the right behavior.

Reply:
Ah nevermind. It looks like the compile definition is used to set test name

https://github.com/ros2/system_tests/blob/394f326f2b3da52d802dad1090cddd8ef6fba5b7/test_rclcpp/test/test_services_client.cpp#L25-L30

The environment variable is set when `custom_launch_test_two_executables()` is called,

Reply:
Ah yeah, that makes sense.